### ISSUE TITLE
AM2R

### ISSUE THREAD

**Issue by Rinnegatamante** [Rinnegatamante/YoYo-Loader-Vita-Compatibility#11]
# Game summary
- Game name: AM2R
- Game version: v.7.5.5

# YoYo Loader summary
- Version: vDebug

# Issues
Works perfectly

# Screenshots
![immagine](https://user-images.githubusercontent.com/9152710/162236573-d23b5320-0c06-486f-afc5-aedf5a341e09.png)
![immagine](https://user-images.githubusercontent.com/9152710/162236583-e7481d67-1d84-4572-afdd-2e691271f113.png)

**Comment by Rinnegatamante** [comment]
Note: Fake Windows as Platform required for controller support. If started without it, game will work with touch and on screen overlay.

**Comment by C1fer** [comment]
The game freezes when going opening the menu

**Comment by SofyUchiha** [comment]
> 

Same for me

**Comment by ChrisHighwind** [comment]
Can confirm, the game can freeze in menus sometimes

**Comment by spirit-in-exile** [comment]
Exiting AM2R via in-game menu crashes the app

**Comment by VNzZzS** [comment]
On my PS Vita AM2R full freeze (crash) some times. This happend when I press start button (open game map).

**Comment by SofyUchiha** [comment]
> The game freezes when going opening the menu

@Rinnegatamante Is there a specific setting that prevents the game from freezing when pressing start to open the map?

**Comment by Ted-bissada** [comment]
I think there is a memory leak or something somewhere in am2r because the same issue would occur in the native vita port

**Comment by VNzZzS** [comment]
I'm testing AM2R for ~1.5h and have 0 crashes. I'm just using start button carefully (one at time, no double clicking or click too often).

**Comment by thipeto** [comment]
so it seems like the "Works perfectly" not apply to AM2R.

Also, sound stops when the first "metroid" is killed

**Comment by VNzZzS** [comment]
About sound. Version with HQ sound works bad (APK size 287mb), version with LQ sound works better (APK size 95mb). Try to use LQ version maybe this is fix your sound problem, thipeto.

**Comment by Ted-bissada** [comment]
something change in commit ([Added support for video playback.](https://github.com/Rinnegatamante/yoyoloader_vita/commit/47ee999893170cde72b90108e8ec0f37d6f95591)) or [Fix wrong texture unit binding in the video player implementation.](https://github.com/Rinnegatamante/yoyoloader_vita/commit/1c1b433720d81c07fda31e9710220be77ec43b66) causes a crash on boot for am2r now

**Comment by Rinnegatamante** [comment]
Are you sure to not be running a Video Playback enabled build?

**Comment by Ted-bissada** [comment]
The build is the one it auto updated to, so the most recent. The video playback option was not checked

**Comment by Rinnegatamante** [comment]
Should be fixed now

**Comment by VNzZzS** [comment]
For now I'm played ~5-7 hours, and my conclusion: unplayable. Two first section was almost perfect but on "Hydro Station"(where game start to lagging a bit ~40fps), I got too many emulator crashes and game full freezees. Unfortunately, that was worse experiense then native version for PS Vita.

**Comment by Rinnegatamante** [comment]
Would actually help to have logs with debug mode rather than just saying "unplayable".

**Comment by Ted-bissada** [comment]
The issue is indeed fixed, once in a while a door transition will get stuck and you need to restart but other than that the game works perfectly. I am quite far into the game and a restart has always bypassed any issues so far.

**Comment by VNzZzS** [comment]
@Rinnegatamante Yes, you right debug logs will be good, but when debug mode activated game runing at 20 fps. Shaders debug mode possible to test if it make sense.

**Comment by beelzebielsk** [comment]
The game runs slow in large areas and areas with water. There are occasional crashes and sometimes the audio stops.

I just got a game crash with debug mode on. What should I do with the logs?

**Comment by beelzebielsk** [comment]
Also, this is really helpful for using debug mode with am2r: if you enter the start menu and back out, the game will run faster, as if debug mode wasn't on. Sometimes it felt even faster than that, oddly. You have to do it once per area you're in.

**Comment by Rinnegatamante** [comment]
I pushed a workaround fix for the Controls menu freeze (may fix also the map one, must be checked). In order to get advantage of it, simply run the apk optimizer feature with latest nightly (be sure to be on 7cffdc or later).

**Comment by lontrafag** [comment]
Game is showing only at quarter of the screen, yyl it's at ver. 0.1 (a3b4ac) 
![2023-04-15-093546-v0 1(a3b4ac)](https://user-images.githubusercontent.com/13691896/232224400-1453a3fe-d530-43d5-b42b-7bb2e2bc8996.png)
![2023-04-15-094950](https://user-images.githubusercontent.com/13691896/232224720-9c4b60bc-a16f-4d8c-80de-173c22f1b30c.png)

**Comment by Rinnegatamante** [comment]
Should be fixed on latest nightly build that should be available in few minutes.
